qt/python/mantidqt/widgets/plotconfigdialog/curvestabwidget/presenter.py

```diff
@@ -56,12 +56,16 @@ def apply_properties(self):
         curve = self.get_selected_curve()
         # Set the curve's new name in the names dict and combo box
         self.set_new_curve_name_in_dict_and_combo_box(curve, view_props.label)
+        self.toggle_errors(self.get_selected_ax(), curve, view_props)
```

Comment:
It looks like we're calling `legend` and calling `relim` twice here. Once in `replot_selected_curve` and then again in `toggle_errorbars`. We probably want to take `relim` and `legend` out of these functions and call them at the end of `apply_properties` and at the end of `FigureErrorsManager.toggle_errorbars_for`?

Reply:
I've refactored the code that decides how to replot the curve into another function, which is now reused from both the `CurvesTabWidgetPresenter` and `FigureErrorsManager`. `apply_properties` now uses that. Additionally the relim/scale/legend is now in `update_limits_and_legend` called from both classes separately